src/imgui/Shortcuts.cc

```diff
@@ -0,0 +1,154 @@
+{
+	static constexpr auto MAX_DESCRIPTION = []{
+		size_t max = 0;
+		size_t imax = 0;
+		for (size_t i = 0; i < allShortcutInfo.size(); ++i) {
+			const auto& info = allShortcutInfo[i];
+			if (info.description.size() > max) {
+				max = info.description.size();
+				imax = i;
+			}
+		}
+		return std::string_view(allShortcutInfo[imax].description);
+	}();
+	return MAX_DESCRIPTION;
+}
+
+// Note: if we forget a row we get a compile error.
+//       if we swap the order of some rows we also get a compile error (when asserts are enabled)
+
+static constexpr auto defaultShortcuts = []{
+    std::array<Shortcuts::Data, ShortcutIndex::NUM_SHORTCUTS> result = {};
+    for (int i = 0; i < ShortcutIndex::NUM_SHORTCUTS; ++i) {
+        auto& all = allShortcutInfo[i];
+        assert(all.index == i); // verify that table is in-order
+        result[i].keyChord = all.keyChord;
+        result[i].local    = all.local;
+        result[i].repeat   = all.repeat;
+    }
+    return result;
+}();
+
+static constexpr auto shortcutNames = []{
+    std::array<std::string_view, ShortcutIndex::NUM_SHORTCUTS> result = {};
+    for (int i = 0; i < ShortcutIndex::NUM_SHORTCUTS; ++i) {
+        result[i] = allShortcutInfo[i].name;
+    }
+    return result;
+}();
+
+static constexpr auto shortcutDescriptions = []{
+    std::array<std::string_view, ShortcutIndex::NUM_SHORTCUTS> result = {};
+    for (int i = 0; i < ShortcutIndex::NUM_SHORTCUTS; ++i) {
+        result[i] = allShortcutInfo[i].description;
+    }
+    return result;
+}();
+
+Shortcuts::Shortcuts(SettingsConfig& config)
+	: shortcuts{}
+{
+	config.setShortcuts(*this);
+	setDefaultShortcuts();
+}
+
+void Shortcuts::setDefaultShortcuts()
+{
+	int index = 0;
+	for (const auto& shortcut : defaultShortcuts) {
+		setShortcut(static_cast<ShortcutIndex>(index++), shortcut.keyChord, shortcut.local, shortcut.repeat);
+	}
+}
+
+const Shortcuts::Data& Shortcuts::getShortcut(ShortcutIndex index)
+{
+	assert(index < ShortcutIndex::NUM_SHORTCUTS);
+	return shortcuts[index];
+}
+
+void Shortcuts::setShortcut(ShortcutIndex index, std::optional<ImGuiKeyChord> keyChord, std::optional<bool> local, std::optional<bool> repeat)
+{
+	assert(index < ShortcutIndex::NUM_SHORTCUTS);
+	if (keyChord) shortcuts[index].keyChord = *keyChord;
+	if (local) shortcuts[index].local = *local;
+	if (repeat) shortcuts[index].repeat = *repeat;
+}
+
+void Shortcuts::setDefaultShortcut(ShortcutIndex index)
+{
+	assert(index < ShortcutIndex::NUM_SHORTCUTS);
+	const auto& shortcut = defaultShortcuts[index];
+	setShortcut(index, shortcut.keyChord, shortcut.local, shortcut.repeat);
+}
+
+std::string_view Shortcuts::getShortcutName(ShortcutIndex index)
+{
+	assert(index < ShortcutIndex::NUM_SHORTCUTS);
+	return shortcutNames[index];
+}
+
+std::optional<ShortcutIndex> Shortcuts::getShortcutIndex(std::string_view name)
+{
+        auto it = ranges::find_if(allShortcutInfo, [&](const auto& p) { return p.name == name; });
```

Comment:
Detail: When the condition in `find_if()` simply selects a single field from a struct it's possible to use `find()` in combination with a "projection":
```auto it = ranges::find(allShortcutInfo, name, &AllShortcutInfo::name);```

BUT this code "materializes" the `allShortcutInfo` table into the executable. That's not really needed. It's sufficient to search in the `shortcutNames` array. (And that can be done with a `find()` without projection).